File: src/main/java/com/derpz/cyberpunkcybernetics/item/ModItems.java
package com.derpz.cyberpunkcybernetics.item;

import com.derpz.cyberpunkcybernetics.CyberpunkCybernetics;
import com.derpz.cyberpunkcybernetics.item.custom.SandevistanArmorItem;
import net.fabricmc.fabric.api.item.v1.FabricItem;
import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.ArmorItem;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.registry.Registries;
import net.minecraft.registry.Registry;
import net.minecraft.util.Identifier;

import java.lang.reflect.Field;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class ModItems {
    // Cyberware
    public static final Item SANDEVISTAN = registerItem("sandevistan", new SandevistanArmorItem(ModArmorMaterials.CYBERWARE, ArmorItem.Type.CHESTPLATE, new FabricItemSettings()));
    public static final Item DEBUG = registerItem("debug", new Item(new FabricItemSettings()));

    // Add more mod items here

    private static Item registerItem(String name, Item item) {
        return Registry.register(Registries.ITEM, new Identifier(CyberpunkCybernetics.MOD_ID, name), item);
    }

    public static void registerModItems() {
        CyberpunkCybernetics.LOGGER.info("Registering Mod Items for " + CyberpunkCybernetics.MOD_ID);
    }

    // Use reflection to get all mod items
    public static Collection<Item> getAllModItems() {
        List<Item> modItems = new ArrayList<>();
        Field[] fields = ModItems.class.getDeclaredFields();
        for (Field field : fields) {
            if (Item.class.isAssignableFrom(field.getType())) {
                try {
                    modItems.add((Item) field.get(null));
                } catch (IllegalAccessException e) {
                    e.printStackTrace();
                }
            }
        }
        return modItems;
    }
}

File: src/main/kotlin/com/derpz/cyberpunkcybernetics/item/ModItems.kt
package com.derpz.cyberpunkcybernetics.item

import com.derpz.cyberpunkcybernetics.CyberpunkCybernetics
import com.derpz.cyberpunkcybernetics.CyberpunkCybernetics.Companion.LOGGER
import com.derpz.cyberpunkcybernetics.item.custom.SandevistanItem
import net.fabricmc.fabric.api.item.v1.FabricItemSettings
import net.fabricmc.fabric.api.itemgroup.v1.FabricItemGroupEntries
import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents
import net.minecraft.item.ArmorItem
import net.minecraft.item.Item
import net.minecraft.item.ItemGroups
import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.util.Identifier

object ModItems {
    //Cyberware
    @JvmField
    val SANDEVISTAN = registerItem("sandevistan", ArmorItem(ModArmorMaterials.CYBERWARE, ArmorItem.Type.CHESTPLATE, FabricItemSettings()))
    val DEBUG = registerItem("debug", Item(FabricItemSettings()))
    private fun addItemToArmorTab(entries: FabricItemGroupEntries) {
        entries.add(SANDEVISTAN)
        /// TODO: 10/2/2023 Redo with for loop to add to custom tab
    }

    private fun registerItem(name: String, item: Item): Item {
        return Registry.register(Registries.ITEM, Identifier(CyberpunkCybernetics.MOD_ID, name), item)
    }

    fun registerModItems() {
        // CyberpunkCybernetics.LOGGER.info("Registering Mod Items for " + CyberpunkCybernetics.MOD_ID);
        LOGGER.info("Registering Mod Items for " + CyberpunkCybernetics.MOD_ID)
        ItemGroupEvents.modifyEntriesEvent(ItemGroups.COMBAT).register { entries ->
            addItemToArmorTab(entries)
        }

    }

    fun getAllModItems(): Collection<Item> {
        val modItems: MutableList<Item> = ArrayList()
        val fields = ModItems::class.java.getDeclaredFields()
        for (field in fields) {
            if (Item::class.java.isAssignableFrom(field.type)) {
                try {
                    modItems.add(field[null] as Item)
                } catch (e: IllegalAccessException) {
                    e.printStackTrace()
                }
            }
        }
        return modItems
    }
}